packages/ld_default.rb

```diff
@@ -2,54 +2,70 @@
 
 class Ld_default < Package
   description 'Select the default ld executable and check for libraries in #{CREW_LIB_PREFIX} first'
-  homepage 'https://github.com/skycocker/chromebrew'
-  version '1.0'
-  source_url 'https://raw.githubusercontent.com/skycocker/chromebrew/93aa4dc4d9198e28d84dcf9d59b3ab6b6a94d92c/README.md'
-  source_sha256 '3ec9c6f7ba6a138d07dfb4a372d562cd3ad558b58b6bd447c2dbb52e16b697a5'
+  homepage 'https://github.com/skycocker/chromebrew/wiki/FAQ'
+  version '1.1'
+  source_url 'file:///dev/null'
+  source_sha256 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'
 
   depends_on 'llvm'
 
   def self.build
-system "cat << 'EOF' > ld_default
+    system "cat << 'EOF' > ld_default
 #!/bin/bash
-cd #{CREW_PREFIX}/bin
-type=\$(file ld | cut -d' ' -f2 | tr -d '\\n')
-if [ \"\$type\" == \"ELF\" ]; then
-  current=\$(find . -inum $(ls -i ld 2> /dev/null | cut -d' ' -f1) 2> /dev/null | fgrep 'ld.' 2> /dev/null | cut -d/ -f2 2> /dev/null)
-else
-  slashes=\$(tr -dc '/' <<<'#{CREW_PREFIX}/bin/ld/' | wc -c)
-  current=\$(tail -1 ld | cut -d'/' -f\${slashes} 2> /dev/null | cut -d' ' -f1 2> /dev/null)
-fi
-echo
-echo \"Current default: \$current\"
-echo
-echo 'Enter the ld default:'
-echo 'b = ld.bfd'
-echo 'g = ld.gold'
-echo 'l = ld.lld'
-echo '0 = Cancel'
-while true; do
-  read default
-  case $default in
-    b)
-      current='ld.bfd'
-      break;;
-    g)
-      current='ld.gold'
-      break;;
-    l)
-      current='ld.lld'
-      break;;
-    0)
-      exit;;
-    *)
-      echo 'Please select from one of the options or enter 0 to cancel.'
-  esac
-done
-echo '#!/bin/bash' > /tmp/ld
-echo '#{CREW_PREFIX}/bin/'$current' --library-path #{CREW_LIB_PREFIX} -rpath #{CREW_LIB_PREFIX} \"\$@\"' >> /tmp/ld
-install -Dm755 /tmp/ld #{CREW_PREFIX}/bin/ld
-rm -f /tmp/ld
+
+  set -e
+
+  mktmp=\$(mktemp -d)
+
+  cd #{CREW_PREFIX}/bin
+
+  type=\$(file ld | cut -d' ' -f2 | tr -d '\\n')
+
+  if [ \"\${type}\" == \"ELF\" ]; then
+    current=\$(find . -inum $(ls -i ld 2>/dev/null | cut -d' ' -f1) 2>/dev/null | fgrep 'ld.' 2>/dev/null)
+  elif [ \"${type}\" == \"symbolic\" ]; then	
+    current=\$(basename $(readlink ld))
+  else
+    current=\$(tail -1 ld | cut -d' ' -f1 2>/dev/null)
+  fi
+
+  echo
+  echo \"  Current default linker: \${current}\"
+  echo
+  echo '  Enter the new default linker:'
+  echo
+  echo '  b = ld.bfd'
+  echo '  g = ld.gold'
+  echo '  l = ld.lld'
+  echo '  0 = Cancel'
+  echo
+
+  while true; do
+    read default
+    case \${default} in
+      b)
+        new='ld.bfd'
+        break;;
+      g)
+        new='ld.gold'
+        break;;
+      l)
+        new='ld.lld'
+        break;;
+      0)
+        exit;;
+      *)
+        echo 'Please select from one of the options or enter 0 to cancel.'
+        echo
+    esac
+  done
+
+  echo '#!/bin/bash' > \${mktmp}/ld
+  echo \"\${new}\"' --library-path #{CREW_LIB_PREFIX} -rpath #{CREW_LIB_PREFIX} \"\$@\"' >> \${mktmp}/ld
```

Comment:
It would be better to use the full path to `${new}`.  You have a lot of faith in your path.

Reply:
The script will look like this: `ld.<ld> --library-path...`
You need to have your Chromebrew installation in the PATH, obviously.

Reply:
I know.  If you use the full path like I had set it up originally, THAT would be an improvement.

Reply:
How? Why would we need that? Users shouldn't yet be using Chromebrew in a directory other than `/usr/local`, anyway.

Reply:
Sorry. I didn't get what you were saying. I think it's better to search the PATH for your linker instead of hardcoding the path (sort of, in the file).

Reply:
The reason you just explained proves my point.

Reply:
Exactly. Instead of using `/usr/local/bin/ld.*`, this uses `ld.*` and has the shell search for the binary in the `$PATH`.

Reply:
`--library-path #{CREW_LIB_PREFIX} -rpath #{CREW_LIB_PREFIX}` are both needed options, though.

Reply:
We are making changes in this directory explicitly.  We don't want to use other linkers by searching `$PATH`.

Reply:
What other linkers? Even if there were any, their `ld` would have higher precedence over this one, depending on the `$PATH`. Example: If someone had Ada in the front of the path, the `ld` from ada would take precedence over this one, anyway.

Reply:
```
chronos@localhost / $ which ld
/usr/local/bin/ld
chronos@localhost / $ PATH=~/.local/bin:$PATH
chronos@localhost / $ which ld
/home/chronos/user/.local/bin/ld
chronos@localhost / $ 
```

Reply:
You are proving my point exactly why you should put the full path to ld.whatever in the ld bash script.  Although somebody could certainly make some other ld command take precedence over this one, what I don't want is some other ld.whatever to take precedence over the ld.whatever in this script.  Do you understand it yet?

Reply:
Yes. It's not going to.

Reply:
A directory later in the PATH can't take control of an earlier directory. If `ld.gold`, `ld.bfd`, and `ld.lld` are in `/usr/local/bin`, then they will always be executed first, unless the user chooses to add a directory to the front of the PATH. If so, then the `ld` in the directory in front of the path will be executed first, anyway, so this `ld` will not be used.

Reply:
I get what you're saying. However, if the user wants to change their PATH for the express reason of having another ld.whatever override this one, shouldn't they be allowed to?

Reply:
I don't think you understand yet.